File: video-series-mono/src/main/java/org/jesperancinha/video/domain/VideoSeries.java
package org.jesperancinha.video.domain;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.jesperancinha.video.data.Genre;

import java.math.BigDecimal;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class VideoSeries {
    @Id
    Long id;
    String name;
    Integer volumes;
    BigDecimal cashValue;
    Genre genre;
}

File: video-series-mono/src/main/kotlin/org/jesperancinha/video/domain/VideoSeries.kt
package org.jesperancinha.video.domain

import org.jesperancinha.video.data.Genre
import java.math.BigDecimal
import jakarta.persistence.Entity
import jakarta.persistence.Id

@Entity
data class VideoSeries (
    @Id
    val id: Long? = null,
    val name: String? = null,
    val volumes: Int? = null,
    val cashValue: BigDecimal? = null,
    val genre: Genre? = null
)